### DISCUSSION TITLE
Pass sensitive information as environment variables

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8507] [Q&A]
Hi everyone,
If I am passing sensitive information to my dagster container as environment variables, how do I pass them as secrets? I would prefer if they did not show in the UI with their values in the launchpad. Thanks!

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1649860880016359?thread_ts=1649860880.016359&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You can specify the config schema to be [`StringSource`](https://docs.dagster.io/_apidocs/config#dagster.StringSource), which allows you to pass a env var name, such as:

```python

from dagster import StringSource, repository, op, job

@op(config_schema={"my_secret": StringSource})
def my_op(context):
    ...


@job(config={"ops": {"my_op": {"config": {"my_secret": {"env": "VERY_SECRET_ENV_VARIABLE"}}}}})
def my_job():
    my_op()
```
